src/dmd/expressionsem.d

```diff
@@ -5364,6 +5364,29 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             return yes();
         }
 
+        if (e.tok2 == TOK.version_ && e.targ && e.targ.ty == Tident && sc && sc._module)
+        {
+            import dmd.cond         : findCondition;
+            import dmd.root.array   : Array;
+            import dmd.access       : checkAccess;
+            // build a list of version id array
+            scope Array!(Identifiers*) idss;
```

Comment:
Scope is useless here. First because Array is a value type anyway. Second because push will still allocate on the heap.

Might as well do away with the foreach and run findCondition directly on each you want to check.  Or alternatively use a nested function instead.